client/core/core.go

```diff
@@ -1272,18 +1273,30 @@ func (c *Core) Run(ctx context.Context) {
 		}
 		symb := strings.ToUpper(unbip(assetID))
 		c.log.Infof("Locking %s wallet", symb)
-		if err := wallet.Lock(); err != nil {
+		if err := walletLockWithTimeout(5*time.Second, wallet); err != nil {
 			c.log.Errorf("Failed to lock %v wallet: %v", symb, err)
 		}
 		wallet.Disconnect()
 	}
 
-	// Let the wallet backends shut down cleanly.
-	c.walletWait.Wait()
-
 	c.log.Infof("DEX client core off")
 }
 
+func walletLockWithTimeout(timeout time.Duration, wallet *xcWallet) error {
+	errChan := make(chan error, 1)
+	go func() {
+		defer close(errChan)
+		errChan <- wallet.Lock()
```

Comment:
It's a little wild to just leave this hung if it takes too long, but I don't see much choice other than trusting the individual wallet implementations (and wiring them with a context or timeout which they could not respect).

I'm fairly comfortable with this sort of abandonment approach in the shutdown paths of `Run` and `Logout` (likely before shutdown).  The other uses are not at the end of `Core` lifetime, but they are equally problematic if there are hangs while, say, clicking the "Lock" button.

Reply:
> giving asset.Wallet.Lock a context or timeout only gets us so far if those backends hang anyway.

That's the crux of it, I think. This is a reasonable move.